frontend/packages/console-app/src/components/pdb/utils/get-pdb-resources.ts

```diff
@@ -10,3 +15,31 @@ export const getPDBResource = (
       resource?.spec?.template?.metadata?.labels || resource?.metadata?.labels || {},
     ),
   );
+
+export const isDisruptionViolated = (pdb: PodDisruptionBudgetKind): boolean => {
+  const { status } = pdb;
+  const conditions = status?.conditions;
+
+  const disruptionNotAllowedCondition = conditions?.find(
+    (condition: PDBCondition) =>
+      condition.type === 'DisruptionAllowed' &&
+      condition.status === K8sResourceConditionStatus.False,
+  );
+  return !!disruptionNotAllowedCondition;
+};
+
```

Comment:
The condition for checking for PDB violation(s) should consider the pod label selector in the applied pod controller(s). If there is a PDB which does not match any pod label selector in the specified namespace, the warning alert is shown. I think that is incorrect. See the screenshots for example


<img width="1817" alt="Screenshot 2024-06-04 at 2 15 50 PM" src="https://github.com/openshift/console/assets/15249132/6fd35a7b-3ae2-420e-ad84-8e840dd6ec36">

<img width="1841" alt="Screenshot 2024-06-04 at 2 16 22 PM" src="https://github.com/openshift/console/assets/15249132/86375598-a634-4f3d-bb44-c394d63f8a3a">

<img width="1856" alt="Screenshot 2024-06-04 at 2 55 41 PM" src="https://github.com/openshift/console/assets/15249132/8a9dc0ed-7256-472f-9368-ae06ee654bc7">

The PDB object `status.expectedPods` will be `0` if it doesn’t match any pod controllers label selector. I think we should include that in the condition.

Reply:
Updated the PR. PTAL.